src/js/actions/layers.js

```diff
@@ -1377,45 +1382,147 @@ define(function (require, exports) {
     };
 
     /**
+     * Helper function to find the right most artboard 
+     * @param {Immutable.List.of<Layer>} artboards
+     * @return {Layer}
+     */
+
+    var _getRightMost = function (artboards) {
+        var layer = artboards.reduce(function (selectedLayer, currentLayer) {
+                if (currentLayer.bounds.right > selectedLayer.bounds.right) {
+                    return currentLayer;
+                } else {
+                    return selectedLayer;
+                }
+            }, artboards.first());
+        return layer;
+    };
+
+    /**
+     * Helper function to get Bounds from specs (template preset)
+     * @param {string} specs, {Immutable.List.of<Layer>} artboards, {Layer} layerRef
+     * @return {Bounds} finalBounds
+     */
+
+    var _getBoundsFromSpecs = function (specs, artboards, layerRef) {
+        var preset,
+            width, 
+            height,
+            rightmostlayer,
+            finalBounds;
+
+        //get the preset template from the specs
+        preset = specs.preset;
+        //find the measurements of that preset 
+        _.each(templates, function (value, index) { 
+            if (templates[index].preset.valueOf() === preset.valueOf()) {
+                height = templates[index].height;
+                width = templates[index].width;
+            } 
+        });
+
+        //if artboards are empty, insert template at first position 
+        if (artboards.isEmpty()) {
+            layerRef = layerLib.referenceBy.current;
+            finalBounds = new Bounds(
+                {top: DEFAULT_ARTBOARD_BOUNDS.top,
+                 bottom: height,
+                 left: DEFAULT_ARTBOARD_BOUNDS.left,
+                 right: width
+                });
+        } else {
+            //find the rightmost artboard and insert the template after this 
+            rightmostlayer = _getRightMost(artboards);
```

Comment:
If you're _only_ using `rightmostLayer` right here in this else branch, it's preferable to declare it here so that it doesn't muddle overall understanding of the function.